File: 2023-2024/JPP/L1/Zadanie_1/Rec/test_rec.c
#include <stdio.h>
#include "Zadanie_1_rec.h"

int main(){
	unsigned int n = 5;
	printf("Factorial of %u is %llu\n",n,factorial(n));
	
	unsigned int a = 35, b = 10;
	printf("GCD of %u and %u is %u\n", a, b, gcd(a,b));

	int c = 55;
	DiophantineSolution solution = diophantine(a, b, c);
	printf("Diophantine equation: %dx + %dy = %d is x = %d, y = %d\n",a, b, c, solution.x, solution.y);

	return 0;
}

File: 2023-2024/JPP/L1/Zadanie_3/Rec/test_rec.py
import zadanie_3_rec

n = 5
k = zadanie_3_rec.factorial(n)
print(f"Factorial of {n} is {k}")

a = 35
b = 10
l = zadanie_3_rec.gcd(a,b)
print(f"GCD of {a} and {b} is {l}")

c = 55 
m = zadanie_3_rec.diophantine(a,b,c)
print(f"Diophantine equation: {a}x + {b}y = {c} is x = {m[0]}, y = {m[1]}")
